nixos/modules/services/monitoring/osquery.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with builtins;
+with lib;
+
+let
+  cfg = config.services.osquery;
+  runtimeValue = flag: default:
+    cfg.config.options.${flag}
+      or cfg.flags.${flag}
+      or default;
+in
+{
+  options.services.osquery = {
+    enable = mkEnableOption (mdDoc "osqueryd daemon");
+
+    config = mkOption {
+      default = { };
+      description = mdDoc ''
+        Configuration to be written to the osqueryd JSON configuration file.
+        To understand the configuration format, refer to https://osquery.readthedocs.io/en/stable/deployment/configuration/#configuration-components.
+      '';
+      example = {
+        options.utc = false;
+      };
+      type = types.attrs;
+    };
+
+    flagfile = mkOption {
+      default = "/etc/osquery/osquery.flags";
+      description = mdDoc ''
+        Path to the flagfile used to provide CLI flags to osqueryd.
+        For more information, refer to https://osquery.readthedocs.io/en/stable/installation/cli-flags/#flagfile.
+      '';
+      type = types.path;
+    };
+    flags = mkOption {
+      default = { };
+      description = mdDoc ''
+        Attribute set of flag names and values to be written to the osqueryd flagfile.
+        For more information, refer to https://osquery.readthedocs.io/en/stable/installation/cli-flags.
+      '';
+      example = {
+        config_refresh = "10";
+      };
+      type = types.attrsOf types.str;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.etc = {
```

Comment:
Why stick these in `/etc`? Couldn't we just have them in the Nix store?

Reply:
Yeah I suppose it could be and that is likely the more NixOS way. I'll look for example of how this is done in other modules.

Reply:
Yeah, much more common IME make a derivation containing your configuration.

See e.g. https://github.com/NixOS/nixpkgs/blob/c650250e5d094f3548f50d484273a5f760a458d7/nixos/modules/services/monitoring/bosun.nix#L8

Reply:
The config is no longer in `etc`.